### ISSUE TITLE
feat(ci): Install StorybookJS with a11y add-on

### ISSUE THREAD

**Issue by Jack-Gledhill** [Jack-Gledhill/jackgledhill.com#325]
## Feature Description

[Storybook](https://storybook.js.org) is a framework-agnostic component sandbox tool that provides a web environment for testing components. 

It has numerous plugins, including [one for Vitest](https://storybook.js.org/docs/writing-tests/integrations/vitest-addon) and another for [accessibility testing](https://storybook.js.org/docs/writing-tests/accessibility-testing). Not only does it provide a UI for sandboxing and documenting components, it can also be used to automate tests via GitHub Actions.

Storybook can even [automatically document components](https://storybook.js.org/docs/writing-docs/autodocs).

Unfortunately, Storybook cannot be used as a comprehensive test suite for the website; it's only able to test **components**, not **pages**. Nevertheless, its features are valuable, as fully automated accessibility testing is not straightforward.

## Proposed Solution

Once installed, stories will need to be created for each component. These stories may present different scenarios in which the component could be used, allowing for a comprehensive overview of the component's capabilities.

A GitHub Action can be created to automatically run Storybook on push, which will run accessibility tests on the components and ensure it adheres to [WCAG AA](https://www.w3.org/WAI/WCAG2AA-Conformance) standards.

## Willingness to Contribute

Remember to check the [Contributing guidelines](/CONTRIBUTING.md) for guidance on how to contribute.

- [x] I'm willing to submit a PR for this feature
- [x] I can help test this feature
- [x] I can provide additional information if needed
